src/ua_types.c

```diff
@@ -1117,11 +1117,18 @@ UA_StatusCode UA_copy(const void *src, void *dst, const UA_DataType *dataType) {
 }
 
 void UA_deleteMembers(void *p, const UA_DataType *dataType) {
+    UA_deleteMembersUntil(p, dataType, 0);
+}
+
+void UA_deleteMembersUntil(void *p, const UA_DataType *dataType, size_t maxOffset) {
     uintptr_t ptr = (uintptr_t)p;
     if(dataType->fixedSize)
         return;
     UA_Byte membersSize = dataType->membersSize;
     for(size_t i=0;i<membersSize; i++) {
+        if(maxOffset >0 && ptr-(uintptr_t)p >=maxOffset){
```

Comment:
You compare `uintptr_t` with a `size_t`.
Imho we could make offset an `uintptr_t` for semantics.
Or we treat the offset as the member until which we want to delete.
`if(i > maxOffset) return;`